butterknife-compiler/src/main/java/butterknife/compiler/BindingClass.java

```diff
@@ -730,6 +734,24 @@ private boolean hasMethodBindings() {
     return false;
   }
 
+  private boolean methodBindingsNeedFinder() {
+    for (ViewBindings viewBindings : viewIdMap.values()) {
+      for (Map<ListenerMethod, Set<MethodViewBinding>> listenerMethodSetMap :
+          viewBindings.getMethodBindings().values()) {
+        for (Set<MethodViewBinding> methodViewBindings : listenerMethodSetMap.values()) {
+          for (MethodViewBinding methodViewBinding : methodViewBindings) {
+            for (Parameter parameter : methodViewBinding.getParameters()) {
+              if (!parameter.getType().toString().equals(VIEW.toString())) {
```

Comment:
This should be using `parameter.requiresCast()` which is the logic the original method uses.